### ISSUE TITLE
Image regression not work as expected

### ISSUE THREAD

**Issue by showkeyjar** [keras-team/autokeras#1631]
### Bug Description

I use autokeras to train a image regression model, and use `reg.evaluate(val_dataset)` the result is about 0.59
but when I use predict function, it return strange results, for example:
label 4.01 predict 2.57
label 3.43 predict 2.86
label 2.33 predict 2.89
the result sames has no relation with label, and regression not work.

### Bug Reproduction
Code for reproducing the bug:

because there is no suitable function in keras or tensorflow to load data for image regression,
I create a custom function called "image_dataset_from_df", this function just load image data from pandas dataframe:
the key code is:
`
def index_df(df,
                    shuffle=True,
                    seed=None):
    global directory
    labels = df["score"].values
    file_paths = [os.path.join(directory, fname) for fname in df["filename"].values]

    if shuffle:
        # Shuffle globally to erase macro-structure
        if seed is None:
            seed = np.random.randint(1e6)
        rng = np.random.RandomState(seed)
        rng.shuffle(file_paths)
        rng = np.random.RandomState(seed)
        rng.shuffle(labels)
    return file_paths, labels

def image_dataset_from_df(...)
    ....
        image_paths, labels = index_df(
        df,
        shuffle=shuffle,
        seed=seed)
    ....
` 
I don't think it affected results.

Data used by the code:

you can use any open face age image dataset.

### Setup Details
Include the details about the versions of:
 - OS type and version:
 - Python: 3.8.5
 - autokeras: 1.0.16
 - keras-tuner:1.0.4
 - scikit-learn:0.24.2
 - numpy:1.19.5
 - pandas:1.3.1
 - tensorflow:2.5.0

### Additional context

none

**Comment by showkeyjar** [comment]
ok, I found problem: because I forgot add normalize operation when predict images.
